The first order reaction where A is converted to B is carried out in a laboratory scale CSTR at different stirrer speeds (in $rpm$), at different temperatures and catalyst particle sizes. The results at a fixed concentration of $A$, $C_{A} = 16.9\times10^{3}~mol/m^{3}$, is shown on the attached figure. The catalyst is porous and $\rho_{cat} = 1200~kg/m^{3}$. It is also known that the total catalyst surface area ($S_{i}$) is $100~m^{2}/g$.

<img src="Tut5.7 data.JPG" width="850" />

In [123]:
from numpy import *
from scipy.optimize import fsolve
from scipy.integrate import odeint
from matplotlib import pyplot as plt
from scipy.interpolate import interp1d
%matplotlib inline

**a) In what region(s) ($rpm$, $T$ and $dp$) is the reaction rate influenced by only internal mass transfer limitations (i.e. no external mass transfer effects)?** ($dp = 5~mm$, $T = 400~K$, $rpm > 900$)

**b) In what region/s is the reaction rate influenced by only external mass transfer
limitations? (i.e. no internal mass transfer effects)**
($400~K$, $d_{p} = 0.3~mm$, $rpm<400$, and $400~K$, $d_{p} = 0.1~mm$, $rpm<200$)

For both questions a) and b):

It is important to recognise that for all the particle diameters at the **higher stirrer speeds**, $C_{A_{b}} = C_{A_{s}}$: External mass transfer is negligible here (when the curve flatlines with rpm). A synonym for this is that external diffusion effects are negligible and for a first order reaction it implies that in:
$$
\begin{align}
\dfrac{1}{k_{eff}^{'}} &= \dfrac{1}{k_{c}a_{m}} + \dfrac{1}{\eta k_{i}^{'}} \nonumber \\
\dfrac{1}{k_{c}a_{m}} &\approx 0\nonumber \\
k_{eff}^{'} &= \eta k_{i}^{'} \nonumber
\end{align}
$$

Also, at $300~K$, $\eta = 1$ for both $d_{p} = 1~mm$ and $d_{p} = 0.5~mm$. We can deduce that since at the point where $C_{A_{s}} = C_{A_{b}}$, the rates for the two diameters are equal.

And at $400~K$, $\eta = 1$ for both diameters shown.

At **lower stirrer speeds** with these small diameters, the difference between the rates (at the same stirrer speed), is due to $k_{c}a_{m} \propto rpm^{m}.d_{p}^{-1}$ and NOT because $\eta < 1$

For $d_{p} = 5~mm$ at $300~K$, $\eta$ is less than one. Even where $C_{A_{s}} = C_{A_{b}}$ (flatline with rpm), the rate at the same temperature and the same $C_{A_{b}}$ is not the same as for the particle sizes where $\eta = 1$. The $\eta$-value here is NOT a function of rpm! So even at lower rpm's where $C_{A_{s}} \neq C_{A_{b}}$ the same value of $\eta$ applies. It will only change if one changes $d_{p}$ and/or Temperature.

-

A packed bed reactor is packed with $100~kg$ of the $5~mm$ particles of this same catalyst ($\epsilon = 0.4$). It is used to process $40~L/min$ of a feed stream with pure component $A$.

In [124]:
CAb    = 16.9e3 #mol/m3
rhocat =   1200 #kg/m3
Si     =  100e3 #m2/kg
R      =  8.314 #J/kg.K

In [125]:
W       =        100 #kg
dp      =     5/1000 #m
Rp      =       dp/2 #m 
epsilon =        0.4
Q       = 40/1000/60 #m3/s

**c) What conversion would be obtained if the reactor is free of any external mass
transfer or mixing effects and is isothermally operated at $400~K$?** ($X = 0.59$)

$$
\begin{align}
\frac{dF_A}{dW}&=-r_{A}^{'}\\
\frac{dF_A}{dW}&=\eta k_{i}^{'}C_A\\
QC_{Ao}\frac{dx}{dW}&=\eta k_{i}^{'}C_{Ao}(1-x)\\
\int^{X}_{0}\frac{dx}{1-x}&=\frac{\eta k_{i}^{'}W}{Q}\\
-\ln(1-X)&=\frac{\eta k_{i}^{'}W}{Q}\\
\end{align}
$$

In [126]:
T = 400 #k
raprime = 0.1 #mol/(kg.s) read off the graph for the 5mm cat particle

In [127]:
keffprime = raprime/CAb
keffprime

5.917159763313609e-06

In [128]:
a = exp(-keffprime*W/Q)  
a

0.4116532276535465

In [129]:
X = 1 - a
X

0.5883467723464535

**d) Calculate the conversion that would be achieved if the temperature of the reactor
(with 100 kg of catalyst) is increased to 500 K and the reactor layout is such that
the rate is still not limited by external mass transfer.** ($\eta = 0.74$ and X $= 0.87$)

1) We know at higher stirrer speeds, $k_{i}^{'}$ does not depend on particle diameter for the same temperature.

2) We also know, because both $d_{p} = 0.3~mm$ and $d_{p} = 0.1~mm$ converge at high stirrer speeds (i.e. both have the same $-r_{A}$), that $\eta = 1$ at $400~K$ for those two. Since $d_{p} = 5~mm$ has a lower $-r_{A}$ at higher stirrer speeds, it's $\eta$ must be less than $1$

3) Using the intrinsic rate equation, $-r_{A} = \eta k_{i}^{'}C_{A_{b}}$, we have two equations:
\begin{align}
0.1355 &= k_{1}^{'}C_{A_{b}} \nonumber \\
0.1000 &= \eta k_{2}^{'}C_{A_{b}} \nonumber \\
k_{1}^{'} &= k_{2}^{'} \nonumber
\end{align}

In [130]:
# therefore, eta for dp=5mm:
eta400 = 0.1/0.1355
eta400

0.7380073800738007

In [131]:
k400 = keffprime/eta400
k400

8.017751479289941e-06

At $300~K$, we can calculate the rate constant where $d_{p}=1~mm$ and $d_{p}=0.5~mm$ converge. $\eta = 1$ at high stirrer speed there as well as both those particle sizes converge.

In [132]:
k300 = 0.018/CAb
k300

1.0650887573964497e-06

In [133]:
Y1 = log(k400)
Y1

-11.733852539573546

In [134]:
Y2 = log(k300)
Y2

-13.752452421997138

In [135]:
X1 = 1/400
X2 = 1/300

In [136]:
#gradient, M is the same for both intrinsic rate lines
#(Y1 - C)/X1 = (Y2 - C)/X2
C = (Y2*X1 - Y1*X2)/(X1 - X2)
C

-5.678052892302772

In [137]:
koprime = exp(C)
koprime

0.0034202114843071532

In [138]:
M = (Y1 - C)/X1
M

-2422.3198589083095

In [139]:
Ea = -R*M
Ea

20139.167306963685

In [140]:
k500 = koprime*exp(-Ea/(R*500))
k500

2.691861212317777e-05

Now we need to calculate $\eta$ at $500~K$ for $d_{p} = 5~mm$

$$\eta_{400~K} = 0.738 = \dfrac{3}{\phi_{1}^{2}}(\phi_{1}coth(\phi_{1}) - 1)$$

In [145]:
phi = 3/eta400 #1st guess
error = 0.1

In [147]:
while error >= 0.0000000001:
    phinew  = ((3/eta400)*((phi/tanh(phi)) - 1))**0.5
    error    = abs(phi - phinew)
    phi      = phinew

In [149]:
phi400 = phi
print(str(phi400))

2.48725078247475


From the Thiele equation:
\begin{align}
\dfrac{\phi_{1}^{2})_{400~K}}{\phi_{1}^{2})_{500~K}} = \dfrac{k_{i}^{'})_{400~K}}{k_{i}^{'})_{500~K}} \nonumber
\end{align}

In [151]:
phi500 = ((phi400**2)*k500/k400)**0.5
phi500

4.5574258693926275

In [155]:
eta500 = (3/(phi500**2))*((phi500/tanh(phi500)) - 1)
eta500

0.5139730031460392

In [156]:
a500 = exp(-eta500*k500*W/Q)  
a500

0.12551675955931763

In [158]:
X500 = 1 - a500 
X500     #0.874 is actually correct if no external mass transfer for a CSTR

0.8744832404406824